plugins/modules/win_domain_computer.ps1

```diff
@@ -146,6 +157,73 @@ Function Add-ConstructedState($desired_state) {
   $result.changed = $true
 }
 
+Function Invoke-OfflineDomainJoin {
+  [CmdletBinding(SupportsShouldProcess=$true)]
+  param(
+    [Parameter(Mandatory=$true)]
+    [System.Collections.IDictionary]
+    $desired_state ,
+
+    [Parameter()]
+    [bool]
+    $ReturnBlob = $false ,
+
+    [Parameter()]
+    [System.IO.FileInfo]
+    $BlobPath
+  )
+
+  End {
+    $dns_domain = $desired_state.dns_hostname -replace '^[^.]+\.'
+
+    $delete = -not $BlobPath
+    if ($delete) {
+      $BlobPath = [System.IO.Path]::GetTempFileName()
+    }
+
+    $arguments = @(
+      '/PROVISION'
+      '/REUSE'  # we're pre-creating the machine normally to set other fields, then overwriting it with this
+      '/DOMAIN'
+      $dns_domain
+      '/MACHINE'
+      $desired_state.sam_account_name.TrimEnd('$')  # this machine name is the short name
+      '/MACHINEOU'
+      $desired_state.ou
+      '/SAVEFILE'
+      $BlobPath.FullName
+    )
+
+    $argstring = Argv-ToString -arguments $arguments
+    $exe = 'djoin.exe'
+    $invocation = "$exe $argstring"
```

Comment:
You may as well put `djoin.exe` as the first value in `$arguments`, that way you can just do `$invocation = Argv-ToString -arguments $arguments`

Reply:
The only reason I ended up keeping it separate was for `ShouldProcess()` but I can use the single argument form of that and just put it together.

Reply:
Oh I didn't realise that, still the executable is always `djoin.exe` so there's no reason why you can't just hardcode that in ShouldProcess().